### PR TITLE
#Cannot uninstall Sitecore if the wdp package was deleted

### LINKED ISSUE #506: Cannot uninstall Sitecore if the wdp package was deleted
**Description:**  
Cannot uninstall Sitecore if the wdp package was deleted.
Encountered in SIM 1.7.1.656, Sitecore 9.3 XP0 (also tested with XM1).

**Scenario:**  
1. Install a new Sitecore instance.
2. Exit SIM.
3. Delete the wdp package from which Sitecore was installed.
4. Run SIM again and try deleting the Sitecore instance.

*Result:*
An error is shown in UI and a related entry is added in logs:
  14:47:55,669 INFO  Error occured while wizard loading
System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Sitecore\Downloads\Sitecore 9.3.0 rev. 003498 (WDP XM1 packages)'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileSystemEnumerableIterator`1.CommonInit()
   at System.IO.FileSystemEnumerableIterator`1..ctor(String path, String originalUserPath, String searchPattern, SearchOption searchOption, SearchResultHandler`1 resultHandler, Boolean checkHost)
   at System.IO.Directory.GetFiles(String path, String searchPattern, SearchOption searchOption)
   at SIM.Sitecore9Installer.Tasker.LoadTasks()
   at SIM.Sitecore9Installer.Tasker..ctor(String unInstallParamsPath)
   at SIM.Tool.Windows.UserControls.Install.Delete9Details.SIM.Tool.Base.Wizards.IWizardStep.InitializeStep(WizardArgs wizardArgs)
   at SIM.Tool.Base.Windows.Wizard.InitializeIWizardStep(Int32 n)
   at SIM.Tool.Base.Windows.Wizard.InitializeStep(Nullable`1 i)
   at SIM.Tool.Base.Windows.Wizard.Initialize()

### PR DESCRIPTION
**Pull request by ViacheslavAfanasyev**
closes #506

### CODE REVIEW

File: src/SIM.Sitecore9Installer/Tasker.cs [hunk 1]
```diff
@@ -323,24 +323,24 @@ public void GenerateScripts()
...
       foreach (JProperty param in doc["ExecSequense"].Children())
       {        
         string realName = GetTaskRealName(param);
-        string taskFile = Directory.GetFiles(FilesRoot, string.Format("{0}.json", realName), SearchOption.AllDirectories)
+        string taskFile = Directory.GetFiles(path, string.Format("{0}.json", realName), SearchOption.AllDirectories)
           .FirstOrDefault();
-        if (!string.IsNullOrEmpty(taskFile))
-        {
-          if (!string.IsNullOrEmpty(deployRoot))
+          if (!string.IsNullOrEmpty(taskFile))
```
Comment (AndreyFilchenkov) [inline src/SIM.Sitecore9Installer/Tasker.cs:334]: Could you please remove extra tab/spaces?

File: src/SIM.Sitecore9Installer/Tasker.cs [hunk 2]
```diff
@@ -323,24 +323,24 @@ public void GenerateScripts()
...
           {
-            InjectLocalDeploymentRoot(taskFile);
-            InjectGlobalDeploymentRoot(deployRoot);
+            if (!string.IsNullOrEmpty(deployRoot))
+            {
+              InjectLocalDeploymentRoot(taskFile);
+              InjectGlobalDeploymentRoot(deployRoot);
+            }
```
Comment (AndreyFilchenkov) [inline src/SIM.Sitecore9Installer/Tasker.cs:340]: Could you please remove extra tab/spaces?

**Review by AndreyFilchenkov — CHANGES_REQUESTED** [review]

File: src/SIM.Sitecore9Installer/Tasker.cs (hunk 1 above)
Comment (ViacheslavAfanasyev) [reply src/SIM.Sitecore9Installer/Tasker.cs:334]: I haven't got which ones? Could you please clarify what you mean? :-)

Comment (ViacheslavAfanasyev) [reply src/SIM.Sitecore9Installer/Tasker.cs]: I've tried to fix but it seems doesn't work ;-)

**Review by IvanSheyenko — APPROVED** [review]

**Pull request merged**